libs/register/feature/src/lib/register.component.ts

```diff
@@ -10,34 +14,356 @@ export class RegisterComponent {
+      this.userDetails.address = this.parentRegisterDetailsForm.value.address;
+      this.userDetails.number = this.parentRegisterDetailsForm.value.phone;
+      this.userDetails.password = this.parentRegisterDetailsForm.value.pass;
+
+      if(this.parentChosen) 
+      {
+        this.userDetails.type = 1;
+        this.userDetails.registered = true;
+      }
+      else
+      {
+        this.userDetails.type = 2;
+      }
+
+      await this.serv.register(this.userDetails)
+      .toPromise()
+      .then(
+        res => {
+          application = res;
+        },
+        error => {
+          console.log("Error has occured with API: " + error);
+        }
+      )
+
+      if(application == "pending")
+      {
+        if(this.parentChosen)
+        {
+          this.openToast("Registration succesfull");
+          this.parentDetails.id = this.userDetails.id;
+          this.parentDetails.medID = this.parentRegisterDetailsForm.value.medAid;
+
+          await this.serv.addParent(this.parentDetails)
+          .toPromise()
+          .then(
+            res => {
+              console.log("The response is:" + res);
+            },
+            error => {
+              console.log("Error has occured with API: " + error);
+            }
+          )
+        }
+        else
+        {
+          this.openToast("Registration succesfull pending approval");
+          this.aupairDetails.id = this.userDetails.id;
+          this.serv.addAuPair(this.aupairDetails)
+          .toPromise()
+          .then(
+            res => {
+              console.log("The response is:" + res);
+            },
+            error => {
+              console.log("Error has occured with API: " + error);
+            }
+          )
+        }
+      }
+      else
+      {
+        this.openToast("Account already exists with email : "+application);
+      }
     }
+  }
+
+  async openToast(message: string)
+  {
+    const toast = await this.toastCtrl.create({
+      message: message,
+      duration: 4000,
+      position: 'top',
+      color: 'primary',
+      cssClass: 'toastPopUp'
+    });
+    await toast.present();
+  }
+
+  async getLocations()
+  {
+    const loc = this.parentRegisterDetailsForm.value.address;
+    
+    const locationParam = loc.replace(' ', '+');
+    const params = locationParam + '&limit=4&format=json&polygon_geojson=1&addressdetails=1';
+
+    await this.http.get('https://nominatim.openstreetmap.org/search?q='+params)
+    .toPromise()
+    .then(
+      data => {
+        const json_data = JSON.stringify(data);
+        const res = JSON.parse(json_data);
+
+        if(json_data === "{}")
+        {
+          return;
+        }
 
-    return 0;
+        const len = res.length;
+        for (let j = 0; j < len && j<4; j++) 
+        { 
+          this.potentialLocations.push(res[j].display_name);
+        }
+        console.log(this.potentialLocations);
```

Comment:
Remove unneeded console logs unless still important for testing purposes please

Reply:
Used for address fucntionality untill custom selector is made

Reply:
Cool, just make note to remove then when done